frame/instrumental-strategy-pablo/src/lib.rs

```diff
@@ -209,7 +254,59 @@ pub mod pallet {
 
 	impl<T: Config> Pallet<T> {
 		#[transactional]
-		fn do_rebalance(_vault_id: &T::VaultId) -> DispatchResult {
+		fn do_rebalance(vault_id: &T::VaultId) -> DispatchResult {
+			let asset_id = T::Vault::asset_id(vault_id)?;
+			let account_id = T::Vault::account_id(vault_id);
+			let pool_id = Self::pools(asset_id).ok_or_else(|| Error::<T>::PoolNotFound)?;
```

Comment:
```suggestion
			let pool_id = Self::pools(asset_id).ok_or(Error::<T>::PoolNotFound)?;
```

My bad, there is no need `ok_or_else` with a closure here.